### ISSUE TITLE
stream does not work properly on opened small files

### ISSUE THREAD

**Issue by FelipeLema** [cdgriffith/puremagic#96]
using the following image (icon-32.png)
![icon-32](https://github.com/user-attachments/assets/1a56a0ce-a365-4994-b532-1dbe5d4c3e37)

the following code fails
```
import puremagic

with open(r"icon-32.png", "rb") as file:
    print(puremagic.magic_stream(file))
```

here's the error

```
Traceback (most recent call last):
  File "/home/…/./playing-with-puremagic", line 4, in <module>
    print(puremagic.magic_stream(file))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/…/.venv/lib/python3.12/site-packages/puremagic/main.py", line 351, in magic_stream
    head, foot = _stream_details(stream)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/…/.venv/lib/python3.12/site-packages/puremagic/main.py", line 229, in _stream_details
    stream.seek(-max_foot, os.SEEK_END)
OSError: [Errno 22] Invalid argument
```

not sure why, but maybe it's because it's trying to `seek()` to a position that's bigger than its size?

**Comment by NebularNerd** [comment]
I think this is related to my issue on #86 

Converting your image to a `.jpg` works, converting it to a `.webp` produces the same errors. It seems to be an issue with certain formats, especially small file sizes. Doing a basic:

```
with open(r"icon-32.png", "rb") as file:
    print(file.read())
```

Shows the file can be read and displayed, so as you guessed it's like it keeps trying to read something not there.

[JPG and WEBP.zip](https://github.com/user-attachments/files/16495327/JPG.and.WEBP.zip)

**Comment by cdgriffith** [comment]
Thanks for the fixes, addressed in 1.27!
